som/learning_rate.go

```diff
@@ -0,0 +1,28 @@
+package som
+
+import "math"
+
+const SmallestRate = 0.01
+
+// LearningRate is a decay function with two available strategies: exponential and linear.
+// Both strategies return the learningRate0 for iteration 0 and SmallestRate
+// for iteration == totalIterations-1
+func LearningRate(iteration, totalIterations int, strategy string, learningRate0 float64) float64 {
```

Comment:
Actually, yes, you're totally right and agree with your suggestion 👍 

One more thing: can you update the `func` docs so that it mentions that if u pass incorrect strategy it will default to particular strategy.